### PR TITLE
Feature/Highlights impact

### PR DESCRIPTION
**Pull request by borisermakof**

### CODE REVIEW

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/RestAnalyticsProvider.java [hunk 1]
```diff
@@ -877,16 +883,22 @@ Call<ResponseBody> setInsightCustomStartTime(
                 "Accept: application/+json",
                 "Content-Type:application/json"
         })
```
Comment (shaykeren) [inline analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/RestAnalyticsProvider.java:885]: this code is not backward compatible, what if user updated the plugin but not the backend ?

**Review by shaykeren — COMMENTED** [review]
I think its not backward  compatible ,I didnt do a full CR please ask Oleh to review this PR, He is more familiar with the plugin ...thanks

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/RestAnalyticsProvider.java (hunk 1 above)
Comment (borisermakof) [reply analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/RestAnalyticsProvider.java:908]: fixed

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java [hunk 2]
```diff
@@ -123,6 +124,10 @@ public interface AnalyticsProvider extends Closeable {
     HttpResponse lowLevelCall(HttpRequest request);
 
     String getHighlightsPerformance(Map<String, Object> queryParams);
-
     String getHighlightsTopInsights(Map<String, Object> queryParams);
+
+    String getHighlightsPerformanceV2(HighlightsRequest request);
+    String getHighlightsTopInsightsV2(HighlightsRequest request);
```
Comment (opoliarush) [inline analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java:130]:
```suggestion

    String getHighlightsTopInsightsV2(HighlightsRequest request);
```

File: src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt [hunk 3]
```diff
@@ -49,4 +72,75 @@ class HighlightsMessageRouterHandler(project: Project) : BaseCommonMessageRouter
...
+        Log.log(logger::trace, project, "getHighlightsPerformance called")
+
+        val payloadQuery = getPayloadQuery(requestJsonNode)
+        if (payloadQuery is ObjectNode) {
+
+            var request = createHighlightsRequest(payloadQuery, "scopedSpanCodeObjectId");
+            val payload = HighlightsService.getInstance(project).getHighlightsPerformanceV2(request)
+
+            val message = SetHighlightsPerformanceMessage(payload)
+            Log.log(logger::trace, project, "sending MAIN/GET_HIGHLIGHTS_PERFORMANCE_DATA message")
```
[the remark this answers is not in the document]
Comment (borisermakof) [reply src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt:88]: fixed

File: src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt [hunk 4]
```diff
@@ -49,4 +72,75 @@ class HighlightsMessageRouterHandler(project: Project) : BaseCommonMessageRouter
...
+        Log.log(logger::trace, project, "getHighlightsTopInsights called")
+
+        val payloadQuery = getPayloadQuery(requestJsonNode)
+        if (payloadQuery is ObjectNode) {
+
+            var request = createHighlightsRequest(payloadQuery, "scopedCodeObjectId");
+            val payload = HighlightsService.getInstance(project).getHighlightsTopInsightsV2(request)
+
+            val message = SetHighlightsTopInsightsMessage(payload)
+            Log.log(logger::trace, project, "sending MAIN/GET_HIGHLIGHTS_TOP_ISSUES_DATA message")
```
[the remark this answers is not in the document]
Comment (borisermakof) [reply src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt:121]: fixed

**Review by kshmidt-digma — APPROVED** [review]

File: analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java (hunk 2 above)
Comment (borisermakof) [reply analytics-provider/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsProvider.java:132]: fixed

File: src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt [hunk 5]
```diff
@@ -49,4 +72,75 @@ class HighlightsMessageRouterHandler(project: Project) : BaseCommonMessageRouter
...
+        Log.log(logger::trace, project, "getHighlightsImpact called")
+
+        val payloadQuery = getPayloadQuery(requestJsonNode)
+        if (payloadQuery is ObjectNode) {
+
+            var request = createHighlightsRequest(payloadQuery, "scopedSpanCodeObjectId");
+            val payload = HighlightsService.getInstance(project).getHighlightsImpact(request)
+
+            val message = SetHighlightsImpactMessage(payload)
+            Log.log(logger::trace, project, "sending MAIN/GET_HIGHLIGHTS_IMPACT_DATA message")
```
[the remark this answers is not in the document]
Comment (borisermakof) [reply src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt]: fixed

File: src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (borisermakof) [reply src/main/kotlin/org/digma/intellij/plugin/ui/highlights/HighlightsMessageRouterHandler.kt]: fixed

**Comment by borisermakof** [comment]
cannot merge because of bug

**Pull request closed without merging**